skbio/io/util.py

```diff
@@ -42,7 +47,17 @@ def _get_filehandle(filepath_or, *args, **kwargs):
     pass through.
     """
     if _is_string_or_bytes(filepath_or):
-        fh, own_fh = open(filepath_or, *args, **kwargs), True
+        if requests.compat.urlparse(filepath_or).scheme in {'http', 'https'}:
+            sess = CacheControl(requests.Session(),
+                                cache=FileCache(gettempdir()))
+            req = sess.get(filepath_or, **kwargs)
+
+            # if the response is not 200, an exception will be raised
+            req.raise_for_status()
+
+            fh, own_fh = BytesIO(req.content), True
```

Comment:
Come to think about it, we should make it the default but still allow
for users to have control of this. So modifying kwargs to set
stream=True if None is provided, that way if someone needs to do
stream=False, they can still change it. That sounds good?

On (May-01-15| 9:01), Jai Ram Rideout wrote:
> > @@ -42,7 +47,17 @@ def _get_filehandle(filepath_or, *args, **kwargs):
> >      pass through.
> >      """
> >      if _is_string_or_bytes(filepath_or):
> > -        fh, own_fh = open(filepath_or, *args, **kwargs), True
> > +        if requests.compat.urlparse(filepath_or).scheme in {'http', 'https'}:
> > +            sess = CacheControl(requests.Session(),
> > +                                cache=FileCache(gettempdir()))
> > +            req = sess.get(filepath_or, **kwargs)
> > +
> > +            # if the response is not 200, an exception will be raised
> > +            req.raise_for_status()
> > +
> > +            fh, own_fh = BytesIO(req.content), True
> 
> Is there a reason we wouldn't want this to be the default?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/scikit-bio/pull/903/files#r29510693